syntax = "proto3";

package fintekkers.models.position;

option java_multiple_files = true;

option java_outer_classname = "MeasureProtos";

enum MeasureProto {
  UNKNOWN_MEASURE = 0;
  DIRECTED_QUANTITY = 1;
  MARKET_VALUE = 2;
  UNADJUSTED_COST_BASIS = 3;
  ADJUSTED_COST_BASIS = 4;

  // The current yield of the security, essentially coupon / current price. The price can be
  // unadjusted cost basis, adjusted cost basis, market value, and so on. This is a bond-centric
  // calculation. For equity securities, the TTM dividends will be used as a coupon equivalent (not
  // currently supported).
  CURRENT_YIELD = 5;

  // The yield if the security is held to maturity. For equities, this will be blank.
  // For bonds this will be calculated as: https://www.wallstreetprep.com/knowledge/yield-to-maturity-ytm/
  // For TIPS, no future inflation adjustments to principal will be included.
  // For FRNs, the assumption is the floating rate doesn't change between now and maturity.
  // In the future, context-overrides will allow customization of these assumptions
  YIELD_TO_MATURITY = 7;
}
